cpp/src/binaryop/binaryop.cpp

```diff
@@ -419,25 +424,50 @@ bool is_same_scale_necessary(binary_operator op)
 std::unique_ptr<column> fixed_point_binary_operation(scalar const& lhs,
                                                      column_view const& rhs,
                                                      binary_operator op,
+                                                     thrust::optional<cudf::data_type> output_type,
                                                      rmm::cuda_stream_view stream,
                                                      rmm::mr::device_memory_resource* mr)
 {
   using namespace numeric;
 
   CUDF_EXPECTS(is_supported_fixed_point_binop(op), "Unsupported fixed_point binary operation");
-  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(),
-               "Both columns must be of the same fixed_point type");
+  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(), "Data type mismatch");
+  CUDF_EXPECTS(op != binary_operator::TRUE_DIV || output_type.has_value(),
+               "TRUE_DIV requires result_type.");
 
-  auto const scale       = compute_scale_for_binop(op, lhs.type().scale(), rhs.type().scale());
-  auto const output_type = is_comparison_binop(op) ? data_type{type_id::BOOL8}  //
-                                                   : data_type{lhs.type().id(), scale};
-  auto out = make_fixed_width_column_for_output(lhs, rhs, op, output_type, stream, mr);
+  auto const scale = op == binary_operator::TRUE_DIV
+                       ? output_type.value().scale()
+                       : compute_scale_for_binop(op, lhs.type().scale(), rhs.type().scale());
+
+  auto const out_type = output_type.value_or(
+    is_comparison_binop(op) ? data_type{type_id::BOOL8} : data_type{lhs.type().id(), scale});
+
+  auto out = make_fixed_width_column_for_output(lhs, rhs, op, out_type, stream, mr);
 
   if (rhs.is_empty()) return out;
 
   auto out_view = out->mutable_view();
 
-  if (lhs.type().scale() != rhs.type().scale() && is_same_scale_necessary(op)) {
+  if (op == binary_operator::TRUE_DIV) {
+    // Adjust scalar so lhs has the scale needed to get desired output data_type (scale)
+    auto const diff         = lhs.type().scale() - rhs.type().scale() - scale;
```

Comment:
```suggestion
    auto const diff         = (lhs.type().scale() - rhs.type().scale()) - scale;
```

Reply:
@nvdbaranec What is the advantage of the extra `()`?

Reply:
I think it's good policy to always be explicit, even for simple precedence cases. Also I think the parentheses aids the reader's brain in actually walking through the semantics of what's going on.

Reply:
Done.